### PR TITLE
[#111] ACA UI updates for V2 platform certificates

### LINKED ISSUE #111: Update ACA Dashboard with Base vs Delta and status fields
1. If the Platform Certificate version is 1.1 or greater then change the Type column o the Platform credential page to indicate if the certificate is a base or delta certificate.
2. Change Credentials on the navigation bar to certificates
3.  On the Platform certificate details page:
    a. Add the Platform Certificate type (base vs. delta)
    b. Update the holder field so it hyper inks to the holder , even if it is a Platform Certificate.
    c. if the componentPlatformCert  exists, then show the serial number of the cert and componentPlatformCert hyperlink to it if it exists in the ACA DB.
    d. If the componentPlatformCertUri exisit then display the uri and hyperlink to it.
    e. If more than 1 Platform Certificate exists (a Platform Certificate chain) then display a count of the number of certificate in the chain and provide a previous next button. Order of the previous and next will depend on the notBefore validity ate of the certificate. 
4. The Hyperlink of the Platform and platform attribute columns on the Validation report page should point to the base Platform Certificate of the Platform Certificate chain.

### PR DESCRIPTION
**Pull request by cyrus-dev**
This issue updates the UI for the newest V2 information for platform attribute certificates. This first push has updates from #111 for items 1, 2 and 3a/b.

However as noted in #111 item 4 will be updated in a separate issue.

Closes #111

### CODE REVIEW

File: HIRS_AttestationCAPortal/src/main/java/hirs/attestationca/portal/util/CertificateStringMapBuilder.java [hunk 1]
```diff
@@ -314,13 +344,35 @@ public static Certificate containsAllChain(
...
+                data.put("numInChain", chainCertificates.size());
+                Collections.sort(chainCertificates, new Comparator<PlatformCredential>() {
+                    @Override
+                    public int compare(final PlatformCredential obj1,
+                            final PlatformCredential obj2) {
+                        return obj1.getBeginValidity().compareTo(obj2.getBeginValidity());
+                    }
+                });
+
+                data.put("chainCredentials", chainCertificates);
```
Comment (chubtub) [inline HIRS_AttestationCAPortal/src/main/java/hirs/attestationca/portal/util/CertificateStringMapBuilder.java]: Should we call this key chainCertificates for consistency?

File: HIRS_AttestationCAPortal/src/main/webapp/WEB-INF/jsp/certificate-details.jsp [hunk 2]
```diff
@@ -321,6 +318,29 @@
...
+                    <c:if test="${not empty initialData.platformType}">
+                        <div class="row">
+                            <div class="col-md-1 col-md-offset-1"><span class="colHeader">Platform Type</span></div>
+                            <div id="platformType" class="col col-md-8">${initialData.platformType}</div>
+                        </div>
+                        <div class="row">
+                            <div class="col-md-1 col-md-offset-1"><span class="colHeader">Platform Chain</span></div>
+                            <div id="platformType" class="col col-md-8">
+                                <span>
+                                    <c:forEach items="${initialData.chainCredentials}" var="credential" varStatus="loop">
```
Comment (chubtub) [inline HIRS_AttestationCAPortal/src/main/webapp/WEB-INF/jsp/certificate-details.jsp]: Change this to initialData.chainCertificates if you change the key in CertificateStringMapBuilder.

File: HIRS_AttestationCAPortal/src/main/java/hirs/attestationca/portal/util/CertificateStringMapBuilder.java (hunk 1 above)
Comment (cyrus-dev) [reply HIRS_AttestationCAPortal/src/main/java/hirs/attestationca/portal/util/CertificateStringMapBuilder.java]: Agreed and updated to certificates.

File: HIRS_AttestationCAPortal/src/main/webapp/WEB-INF/jsp/certificate-details.jsp (hunk 2 above)
Comment (cyrus-dev) [reply HIRS_AttestationCAPortal/src/main/webapp/WEB-INF/jsp/certificate-details.jsp]: Updated to reflect the change.

**Pull request merged**